### PR TITLE
Add analytics flow

### PR DESCRIPTION
**Pull request by ShivamSinghania**
**Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).**
- [x] REQUIRED: Do you have an Issue **assigned to you** within the v1 milestone for this PR?  Put the Issue number here: https://github.com/WorldHealthOrganization/app/issues/674 , https://github.com/WorldHealthOrganization/app/issues/640
- [x] Provided detailed pull request description and a succinct title (consider template below for guidance).
- [x] Tested your changes, especially after any code review iterations.
- [x] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [x] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [content/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/content/credits.yaml) file (if you want it to be).

After all boxes above are checked, request and receive an Approved review from any team member knowledgable in the area (TODO team member list).  Once approved, the team member will assign your review to a Committer or use the `needs-merge` label.

## What does this PR accomplish?
- Enables analytics after onboarding.
- Adds analytics preference to shared preferences & code to either reset & disable analytics or enable it.
- Uses analytics preference from shared preferences to reflect setting's button state correctly.
- Remove `kAnalyticsAllowed` from constants.dart & corresponding code from main.dart as it's not required anymore.

Closes https://github.com/WorldHealthOrganization/app/issues/674
Closes https://github.com/WorldHealthOrganization/app/issues/640

## Did you add any dependencies?
No.

## How did you test the change?
Tested in Android Emulator along with breakpoint debugging.
Not tested on iOS.

### CODE REVIEW

File: client/flutter/lib/api/user_preferences.dart [hunk 1]
```diff
@@ -21,8 +22,25 @@ class UserPreferences {
     return (await SharedPreferences.getInstance())
         .setBool(UserPreferenceKey.OnboardingCompleted.toString(), value);
   }
+
+  Future<bool> getAnalyticsEnabled() async {
+    return (await SharedPreferences.getInstance())
```
Comment (advayDev1) [inline client/flutter/lib/api/user_preferences.dart]: nice good default!

File: client/flutter/lib/pages/settings_page.dart [hunk 2]
```diff
@@ -18,52 +17,71 @@ class SettingsPage extends StatefulWidget {
...
+                    fontSize: 25.0,
+                    fontWeight: FontWeight.w700,
+                  ),
+                ),
+                trailing: CupertinoSwitch(
+                  activeColor: Constants.primaryColor,
+                  value: _switchValue ?? false,
+                  onChanged: (newValue) {
+                    _toggleAnalytics(newValue);
+                    setState(() {
```
Comment (advayDev1) [inline client/flutter/lib/pages/settings_page.dart]: can you make this wait for _toggleAnalytics to finish?  then we show its off only after its off

File: client/flutter/pubspec.lock [hunk 3]
```diff
@@ -305,7 +319,7 @@ packages:
       name: test_api
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.2.15"
+    version: "0.2.11"
```
Comment (epicfaace) [inline client/flutter/pubspec.lock]: I don't think this was supposed to decrease?

File: client/flutter/lib/pages/settings_page.dart (hunk 2 above)
Comment (ShivamSinghania) [reply client/flutter/lib/pages/settings_page.dart]: Can do, but it's not required. Also if it's waiting long, button won't change state upon user tap.
Should I change it ?

File: client/flutter/pubspec.lock (hunk 3 above)
Comment (ShivamSinghania) [reply client/flutter/pubspec.lock]: No, it wasn't intended.

**Comment by epicfaace** [comment]
@advayDev1 do we want to go ahead and revert this change from #621 (https://github.com/WorldHealthOrganization/app/pull/621#pullrequestreview-384280655)? (setting IS_ANALYTICS_ENABLED to false in GoogleService-Info.plist). This appears to be a global flag that disables analytics regardless of whether you enabled it in Firebase.

(note that because firebase_analytics_collection_enabled is set to false in AndroidManifest.xml, analytics will not be collected at the beginning regardless)

**Comment by ShivamSinghania** [comment]
@epicfaace can you test this on iOS ?

File: client/flutter/lib/pages/settings_page.dart (hunk 2 above)
Comment (advayDev1) [reply client/flutter/lib/pages/settings_page.dart]: Yes plesae change it

**Comment by advayDev1** [comment]
Here's a possible test plan:
````

First run, no analytics sent until TOS screen ACKed
After TOS screen ACK-ed, analytics is sent on same run of app
Close the app
After TOS screen ACK-ed, analytics is sent on second run of app, and settings ui shows opt-in
Then, if you opt-out analytics is not sent after opting out
Close the app
On third run, opt-out setting is preserved in UI and no analytics is sent
Then if you opt-back-in to analytics, it is sent after doing so
Close the app
On fourth run, opt-in setting is preserved in UI and analytics is sent


````

**Comment by advayDev1** [comment]
Once you have confirmed the test cases on iOS and android please merge @epicfaace

**Comment by advayDev1** [comment]
> @advayDev1 do we want to go ahead and revert this change from #621 ([#621 (review)](https://github.com/WorldHealthOrganization/app/pull/621#pullrequestreview-384280655)) in this PR? (setting IS_ANALYTICS_ENABLED to false in GoogleService-Info.plist). This appears to be a global flag that disables analytics regardless of whether you enabled it in Firebase. Otherwise, this PR won't actually enable analytics because of that flag.
> 
> (note that because firebase_analytics_collection_enabled is set to false in AndroidManifest.xml, analytics will not be collected at the beginning regardless)

@epicfaace if the test plan above is confirmed to work with IS_ANALYTICS_ENABLED=true, yes.

File: client/flutter/pubspec.lock [hunk 4]
```diff
@@ -146,7 +160,7 @@ packages:
       name: intl
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.16.0"
```
Comment (epicfaace) [inline client/flutter/pubspec.lock]: Was this supposed to change?

Comment (ShivamSinghania) [reply client/flutter/pubspec.lock]: Again, not intended.

File: client/flutter/pubspec.lock [hunk 5]
```diff
@@ -146,7 +160,7 @@ packages:
...
+    version: "0.16.1"
```
Comment (epicfaace) [inline client/flutter/pubspec.lock]:
```suggestion
    version: "0.16.0"
```

**Pull request merged**